lib/sign.js

```diff
@@ -17,11 +18,51 @@ var xpi = require("./xpi");
 
 var AMO_API_PREFIX = require("./settings").AMO_API_PREFIX;
 
+function getXpiInfo(options) {
+  options = _.assign({
+    addonDir: null,
+    xpiPath: null,
+  }, options);
+
+  if (!options.xpiPath) {
+    return utils.getManifest({addonDir: options.addonDir})
+      .then(function(manifest) {
+        return getXpiInfoFromManifest(manifest);
+      });
+  } else {
+    return utils.extractXPI(options.xpiPath)
```

Comment:
it could but when I get around to implementing the RDF parsing, the else branch would have parsed the XPI twice which seems wasteful -- i.e. a non-manifest XPI will require a call to `extractXPI()`